cmd/nodepool/aws/create.go

```diff
@@ -60,6 +62,16 @@ func (o *AWSPlatformCreateOptions) UpdateNodePool(ctx context.Context, nodePool
 		}
 	}
 
+	client, err := util.GetClient()
+	if err != nil {
+		return err
+	}
+
+	// Validate HostedCluster payload can support the NodePool CPU type
```

Comment:
UpdateNodePool is not meant to run validations. Can you please be consistent with what you are doing in azure and the structure pattern introduced by
https://github.com/openshift/hypershift/pull/4018
https://github.com/openshift/hypershift/pull/4212

i.e in NewCreateCommand something like
```
	cmd.RunE = func(cmd *cobra.Command, args []string) error {
		// Validate HostedCluster payload can support the NodePool CPU type
		if err = core.ValidateHostedClusterPayloadSupportsNodePoolCPUType(ctx, client, hcluster.Name, hcluster.Namespace, nodePool.Spec.Arch); err != nil {
			return err
		}

		return coreOpts.CreateRunFunc(platformOpts)(cmd, args)
	}

	return cmd
```

After merging this (or any) PR the cli based code can only be better, introducing more tech debt is not acceptable

Reply:
Also why is this being added to this two particular platforms instead of to the core validation?

Reply:
There was no core validation for NodePool. I added a function for this and removed the call to validateHostedClusterPayloadSupportsNodePoolCPUType in those two platforms.